services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/tormaystarkastelu/TormaystarkasteluTulos.kt

```diff
@@ -46,13 +48,37 @@ data class TormaystarkasteluTulos(
             when (max) {
                 linjaautoliikenneindeksi -> IndeksiType.LINJAAUTOLIIKENNEINDEKSI
                 raitioliikenneindeksi -> IndeksiType.RAITIOLIIKENNEINDEKSI
-                autoliikenneindeksi -> IndeksiType.AUTOLIIKENNEINDEKSI
+                autoliikenne.indeksi -> IndeksiType.AUTOLIIKENNEINDEKSI
                 else -> IndeksiType.PYORALIIKENNEINDEKSI
             }
         LiikennehaittaindeksiType(max, type)
     }
 }
 
+@Schema(description = "Car traffic nuisance index and classification")
+@JsonIgnoreProperties(ignoreUnknown = true)
+data class Autoliikenneluokittelu(
+    @JsonView(ChangeLogView::class) val haitanKesto: Int,
+    @JsonView(ChangeLogView::class) val katuluokka: Int,
+    @JsonView(ChangeLogView::class) val liikennemaara: Int,
+    @JsonView(ChangeLogView::class) val kaistahaitta: Int,
+    @JsonView(ChangeLogView::class) val kaistapituushaitta: Int,
+) {
+    @get:JsonView(ChangeLogView::class)
+    val indeksi: Float by lazy {
+        if (katuluokka == 0 || liikennemaara == 0) {
+            0.0f
+        } else {
```

Comment:
Isn't this jumping the gun a bit? There's a separate task for this. Why implement it here?

Besides, the index should be zero when both are zeroes.

There are no tests for this logic.

Reply:
The logic is wrong, you're right about that. I checked it only from HAI-2600 where it was either one. I fixed the phrasing there. I thought to implement it here since it only needed this single if-clause. But, if we use the stored value in database and not this lazyly calculated value, then this calculation functionality would be placed elsewhere. The old implementation of using a map of weights was a bit clumsy imo. I will think of something else...

Reply:
Sorry for misphrasing the ticket description.

Yes, I agree the old implementation with the LuokitteluType map was too complex. This simple calculation is a lot nicer.

Reply:
I left this if-clause in the new function which calculates the index, corrected its logic and added tests that checks this logic.